app/misc/git.ts

```diff
@@ -1343,3 +1375,79 @@ function fetchFromOrigin() {
     displayModal("No Path Found.")
   }
 }
+
+//This function gets the number of commits made on a local repo, either pushed or not
+//The value is stored in total_commit
+@Injectable()
+export function countLocalCommits() {
+  var walker = null;
+
+  Git.Repository.open(repoFullPath)
+      .then(function (repo) {
+        walker = repo.createRevWalk();
+        return repo.getHeadCommit();
+      })
+      .then(function (commit) {
+        walker.sorting(Git.Revwalk.SORT.REVERSE);
+        walker.push(commit.id());
+        walker.sorting
+        walker.pushHead();
+        return walker.getCommits(1000)
+      })
+      .then(function (commits) {
+        total_commit = commits.length;
+
+      })
+
+}
+
+//This function counts the total of pushed commits made on a remote repo by walking through the history
+//The number of commits are grabbed using an async function 
+//The number of commits is stored in commit_diff
+@Injectable()
+export function getAllPushedCommits() {
+  clearModifiedFilesList();
+  var repos;
+  var allCommits = [];
+  var aclist = [];
+
+  Git.Repository.open(repoFullPath)
+      .then(function (repo) {
+        repos = repo;
+        return repo.getReferences(Git.Reference.TYPE.LISTALL)
+      })
+      .then(function (refs) {
+        var count = 0;
+        async.whilst(function () {
```

Comment:
I don't know if you've rebased to master lately but I updated the Async library which changes whilst a little. Take a look at the getallcommits function in master to see the new syntax

Reply:
thanks I have rebased with master and the function is updated now.